src/CommunityToolkit.Aspire.Hosting.Java/JavaAppHostingExtension.Executable.cs

```diff
@@ -0,0 +1,160 @@
+            {
+                State = new("Building Maven project", KnownResourceStates.Starting)
+            }).ConfigureAwait(false);
+
+            logger.LogInformation("Building Maven project");
+
+            var isWindows = RuntimeInformation.IsOSPlatform(OSPlatform.Windows);
+
+            var mvnw = new Process
+            {
+                StartInfo = new ProcessStartInfo
+                {
+                    FileName = isWindows ? "cmd" : "sh",
+                    Arguments = isWindows ? $"/c {mavenOptions.Command} {string.Join(" ", mavenOptions.Args)}" : $"./{mavenOptions.Command} {string.Join(" ", mavenOptions.Args)}",
+                    WorkingDirectory = mavenOptions.WorkingDirectory ?? javaAppResource.WorkingDirectory,
+                    RedirectStandardOutput = true,
+                    RedirectStandardError = true,
+                    CreateNoWindow = true,
+                    UseShellExecute = false,
+                }
+            };
+
+            mvnw.OutputDataReceived += async (sender, args) =>
+            {
+                if (!string.IsNullOrWhiteSpace(args.Data))
+                {
+                    await notificationService.PublishUpdateAsync(javaAppResource, state => state with
+                    {
+                        State = new(args.Data, KnownResourceStates.Starting)
+                    }).ConfigureAwait(false);
+
+                    logger.LogInformation("{Data}", args.Data);
+                }
+            };
+
+            mvnw.ErrorDataReceived += async (sender, args) =>
+            {
+                if (!string.IsNullOrWhiteSpace(args.Data))
+                {
+                    await notificationService.PublishUpdateAsync(javaAppResource, state => state with
+                    {
+                        State = new(args.Data, KnownResourceStates.FailedToStart)
+                    }).ConfigureAwait(false);
+
+                    logger.LogError("{Data}", args.Data);
+                }
+            };
+
+            mvnw.Start();
+            mvnw.BeginOutputReadLine();
+            mvnw.BeginErrorReadLine();
+
+            await mvnw.WaitForExitAsync(ct).ConfigureAwait(false);
+
+            if (mvnw.ExitCode != 0)
+            {
+                await notificationService.PublishUpdateAsync(javaAppResource, state => state with
+                {
+                    State = new($"mvnw exited with {mvnw.ExitCode}", KnownResourceStates.FailedToStart)
+                }).ConfigureAwait(false);
+
+                throw new InvalidOperationException($"mvnw build failed with exit code {mvnw.ExitCode}");
+            }
+        });
+
+        return builder;
+    }
+
+    private static IResourceBuilder<JavaAppExecutableResource> WithJavaDefaults(
+        this IResourceBuilder<JavaAppExecutableResource> builder,
+        JavaAppExecutableResourceOptions options) =>
+        builder.WithOtlpExporter()
```

Comment:
Does this work without making https work?

Reply:
Without HTTPS on the dashboard?

Reply:
Yea does this work right now?

Reply:
I assume so 🤣